pyiron_base/state/settings.py

```diff
@@ -372,10 +379,48 @@ def _get_config_from_environment(self) -> Union[Dict, None]:
             elif k in self.environment_credential_map:
                 config[self.environment_credential_map[k]] = v
         config = self._fix_boolean_var_in_config(config=config)
-        config = self._add_credentials_from_file(config)
         return config if len(config) > 0 else None
 
-    def _add_credentials_from_file(self, config: dict) -> Dict:
+    def _add_credentials_from_file(self) -> Dict:
+        if (
+            "credentials_file" in self._configuration
+            and self._configuration["credentials_file"] is not None
+        ):
+            credential_file = self._configuration["credentials_file"]
+
+            # This gets all the entries in the credential file with the headers
+            # The key-value pairs in the [DEFAULT] section are added everywhere!
+            parser = ConfigParser(inline_comment_prefixes=(";",), interpolation=None)
+            parser.read(credential_file)
+            credentials = {}
+            for sec_name, section in parser.items():
+                credentials_w = {}
+
+                for k, v in section.items():
+                    if k.upper() in self.file_credential_map:
+                        credentials_w[self.file_configuration_map[k.upper()]] = v
+                    else:
+                        credentials_w[k.lower()] = v
```

Comment:
This is a little weird looking, but I guess it's something like this: there is a str:str map with all upper-case keys and lower-case values, and if you can find the credential key in there, you store it at the remapped key; else you force it to be lower and store it regardless.

Could we extract this into its own function to make the current scope more readable? Something like this maybe:

```
credentials_w[self._get_remapped_credential_key(k)] = v
```

and 

```
def _get_remapped_credential_key(self, k):
    """
    A docstring like my comment if my comment was correct, maybe with an explanation of 
    why we would want to keep a (k,v) even if k isn't in our nice map
    """
    if k.upper() in self.file_credential_map:
        return self.file_configuration_map[k.upper()]
    else:
        return k.lower()
```

Actually, now I'm a little confused why there are two separate maps being used here! I didn't even notice that until I sat down to re-write the method 😝 So it is maybe even more helpful than I thought to extract it as a method and include a little docstring helping devs understand why we're doing what we're doing here.